### PR TITLE
Fix rootless_podman when /home has different filesystem

### PR DESCRIPTION
**Pull request by jlausuch**
- Remove check for bsc#1182874 as it's a closed bug.
- Move podman installation at the beginning
- Handle situation when /home is not btrfs   

Related comment: https://bugzilla.suse.com/show_bug.cgi?id=1196751#c44
Verification runs:
[TW](https://openqa.opensuse.org/tests/2296000)
[TW JeOS](https://openqa.opensuse.org/tests/2296001)
[15-SP3]: [osd](https://openqa.suse.de/tests/8539044) [local](http://fromm.arch.suse.de/tests/6503)
[15-SP3 JeOS]:  [osd](https://openqa.suse.de/tests/8539041)  [local](http://fromm.arch.suse.de/tests/6504)
[15-SP2]: [osd](https://openqa.suse.de/tests/8539043)    [local](http://fromm.arch.suse.de/tests/6505)

### CODE REVIEW

**Review by pdostal — APPROVED** [review]
All good :+1:

**Review by Vogtinator — CHANGES_REQUESTED** [review]
This is a product bug

File: tests/containers/rootless_podman.pm [hunk 1]
```diff
@@ -100,10 +102,17 @@ sub run {
...
+    # but if the home partition is not btrfs podman commands will fail with
+    # Error: "/home/bernhard/.local/share/containers/storage/btrfs" is not on a btrfs filesystem
+    if (script_output("podman info", proceed_on_failure => 1) =~ m/prerequisites for driver not satisfied/) {
+        record_soft_failure("bsc#1197093 - /home partition is in different filesystem");
+        record_info('partitions', script_output('lsblk -f'));
+        record_info('storage.conf', script_output('cat /etc/containers/storage.conf'));
+        assert_script_run('sed \'s/^driver.*/driver \= \"overlay\"/\' -i /etc/containers/storage.conf');
+        assert_script_run("rm -rf /home/$user/.local/share/containers/");
+        assert_script_run('podman info');
         return 1;
```
Comment (mloviska) [inline tests/containers/rootless_podman.pm:116]: Should the test continue after it changes the driver, or it should exit anyway ?

Comment (mloviska) [reply tests/containers/rootless_podman.pm:116]: ah the return on L#96 is gone, please the returns in `softfail_on_bsc1197093` as they are meaningless now.

Comment (jlausuch) [reply tests/containers/rootless_podman.pm:116]: yes, done :)

**Review by grisu48 — APPROVED** [review]
LGTM :rocket:

File: tests/containers/rootless_podman.pm [hunk 2]
```diff
@@ -91,7 +94,21 @@ sub run {
...
     select_console "user-console";
 
-    return if softfail_and_skip_on_bsc1182874();
+    # By default the storage driver is set to btrfs if /var is in btrfs
+    # but if the home partition is not btrfs podman commands will fail with
+    # Error: "/home/bernhard/.local/share/containers/storage/btrfs" is not on a btrfs filesystem
+    if (script_output("podman info 2>&1", proceed_on_failure => 1) =~ m/prerequisites for driver not satisfied/) {
+        record_soft_failure("bsc#1197093 - /home partition is in different filesystem");
+
+        $self->select_serial_terminal;
```
Comment (Vogtinator) [inline tests/containers/rootless_podman.pm:103]: A better workaround might be to `mkdir -p ~/.config/containers/ && sed \'s/^driver.*/driver \= \"overlay\"/\' /etc/containers/storage.conf > .config/containers/storage.conf`. That doesn't need root.

Comment (jlausuch) [reply tests/containers/rootless_podman.pm]: This way I get a new error:
```
podman info
Error: error creating runtime static files directory: mkdir /var/lib/containers/storage: permission denied
```
http://fromm.arch.suse.de/tests/5980#step/rootless_podman/181

File: tests/containers/rootless_podman.pm [hunk 3]
```diff
@@ -28,9 +28,14 @@ use bootloader_setup 'add_grub_cmdline_settings';
 use Utils::Architectures;
 use transactional 'process_reboot';
 
+
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/containers/rootless_podman.pm]: removed

File: tests/containers/rootless_podman.pm [hunk 4]
```diff
@@ -89,7 +88,24 @@ sub run {
...
+        record_soft_failure("bsc#1197093 - /home partition is in different filesystem");
+        record_info('partitions', script_output('lsblk -f'));
+        record_info('storage.conf', script_output('cat /etc/containers/storage.conf'));
+        assert_script_run("mkdir -p ~/.config/containers");
+        assert_script_run('cp /etc/containers/storage.conf ~/.config/containers/storage.conf');
+        my $file = '~/.config/containers/storage.conf';
+        assert_script_run("sed -i \'s/^driver.*/driver \= \"overlay\"/\' $file");
+        #assert_script_run('sed -i \'s/^driver.*/driver \= \"overlay\"/\' ~/.config/containers/storage.conf');
+        assert_script_run("sed -i \'s/^runroot.*/runroot \= \"\\/run\\/user\\/1000\\/containers\"/\' $file");
+        assert_script_run("sed -i \'s/^graphroot.*/graphroot = \"\\/home\\/$user\\/.local\\/share\\/containers\\/storage\"/\' $file");
```
Comment (pdostal) [inline tests/containers/rootless_podman.pm:104]:
```suggestion
        file_content_replace($file, '^driver:*' => 'driver = "overlay"');
        file_content_replace($file, '^runroot.*' => 'runroot ="/run/user/1000/containers"');
        file_content_replace($file, '^graphroot.*' => 'graphroot ="/home/' . $user . /.local/share/containers/storage"');
```

**Review by pdostal — APPROVED** [review]
Sorry I didn't find this sooner.

File: tests/containers/rootless_podman.pm (hunk 4 above)
Comment (jlausuch) [reply tests/containers/rootless_podman.pm:104]: ok, will try this. Thanks.

**Comment by jlausuch** [comment]
@Vogtinator PR updated and VRs green/orange.

**Comment by jlausuch** [comment]
I would like to move this forward as current OSD jobs for 15-SP3 are not testing rootless podman feature, just soft-failing the test. 
If there are additional comments, I can address them in another PR. 
Thanks for the reviews.

**Pull request merged**